samples/snippets/src/main/java/com/example/storage/object/DeleteObject.java

```diff
@@ -38,16 +39,18 @@ public static void deleteObject(String projectId, String bucketName, String obje
       System.out.println("The object " + objectName + " wasn't found in " + bucketName);
       return;
     }
-
-    // Optional: set a generation-match precondition to avoid potential race
-    // conditions and data corruptions. The request to upload returns a 412 error if
-    // the object's generation number does not match your precondition.
-    Storage.BlobSourceOption precondition =
-        Storage.BlobSourceOption.generationMatch(blob.getGeneration());
-
-    storage.delete(bucketName, objectName, precondition);
-
-    System.out.println("Object " + objectName + " was deleted from " + bucketName);
+    BlobId idWithGeneration = blob.getBlobId();
+    // Deletes the blob specified by its id. When the generation is present and non-null it will be
+    // specified in the request.
+    // If versioning is enabled on the bucket and the generation is present, only that version of
+    // the object will be deleted.
+    // If instead you want to delete the current version, the generation should be dropped by
+    // performing the following before calling delete.
+    // BlobId idWithoutGeneration =
+    //    BlobId.of(idWithGeneration.getBucket(), idWithGeneration.getName());
```

Comment:
Should we specify that in this case, `storage.delete(idWithoutGeneration)` would need to be called instead of `storage.delete(idWithGeneration)`?